src/wolfio.c

```diff
@@ -999,9 +1014,83 @@ int wolfIO_Send(SOCKET_T sd, char *buf, int sz, int wrFlags)
     sent = (int)SEND_FUNCTION(sd, buf, (size_t)sz, wrFlags);
     sent = TranslateReturnCode(sent, (int)sd);
 
+    if (sent < 0) {
+        int last_err = wolfSSL_LastError(sent);
+        if ((last_err == SOCKET_EWOULDBLOCK)
+#if SOCKET_EWOULDBLOCK != SOCKET_EAGAIN
+            || (last_err == SOCKET_EAGAIN)
+#endif
+#ifdef SOCKET_ETIMEDOUT
+            || (last_err == SOCKET_ETIMEDOUT)
+#endif
+            )
+        {
+            return SOCKET_NODATA;
+        }
+    }
+
+    return sent;
+}
+
```

Comment:
This is handled in `EmbedSend` in `TranslateIoError`. I don't think it is necessary in this PR.

Reply:
fixed by refactor